week2/task7/Program.cs

```diff
@@ -0,0 +1,81 @@
+﻿// See https://aka.ms/new-console-template for more information
+Console.WriteLine("Hello, World!");
+var signaler = new Signaler();
+signaler.AddTime(new JupiterTime(2, 00));
+signaler.AddTime(new JupiterTime(4, 00));
+signaler.AddTime(new JupiterTime(6, 00));
+signaler.Inform();
+// We woke up at 4:21
+signaler.Check(new JupiterTime(1, 17));
+public class JupiterTime
+{
+    private int _hours, _minutes;
+    public JupiterTime(int hours, int minutes)
+    {
+        Hours = hours;
+        Minutes = minutes;
+    }
+    public int Hours
+    {
+        get
+        {
+            return _hours;
+        }
+        set
+        {
+            _hours = value;
+        }
+    }
+    public int Minutes
+    {
+        get
+        {
+            return _minutes;
+        }
+        set
+        {
+            _minutes = value;
+        }
+    }
+
+    public override string ToString()
+    {
+        return "Time: " + _hours + ":" + _minutes;
+    }
+
+}
+
+public class Signaler
+{
+    List<JupiterTime> SignalerTime = new List<JupiterTime>();
+    public void AddTime(JupiterTime time)
+    {
+        SignalerTime.Add(time);
+    }
+    public void Inform()
+    {
+        if (SignalerTime.Count == 0)
+        {
+            Console.WriteLine("No timers added yet");
+        }
+        else
+        {
+            foreach (var time in SignalerTime)
+            {
+                Console.WriteLine($"({time.Hours:00}:{time.Minutes:00})");
+            }
+        }
+    }
+    public void Check(JupiterTime time)
+    {
+        foreach (var timer in SignalerTime)
+        {
+            if (time.Hours > timer.Hours || (time.Hours == timer.Hours && time.Minutes > timer.Minutes))
+            {
+                Console.WriteLine($"({timer.Hours:00}:{timer.Minutes:00})");
+            }
+
+        }
+        Console.WriteLine("No signals needed to be sent yet");
```

Comment:
This line will always be printed since there is nothing to skip it if necessary. this would mean that you can end up with something like this: 
`
`(02:00)
(04:00)
(06:00)
No signals needed to be sent yet`

The above is not necessarily correct. You can avoid this by, for example, creating a new list with the timers that are missed, and then you can have an if else to check if there are any times missed or not. 

// create new List<JupiterTime>
// foreach SignalerTime check if it late
    // if it is late, add to the list
 
 Outside the loop 
 // if any items in the list, print the times
 // else print no signals missed

Reply:
let me know if the above makes sense or if I should explain in more detail 😊